core/controllers/android_e2e_config_test.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for the admin page."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from constants import constants
+from core.domain import platform_parameter_domain
+from core.domain import topic_fetchers
+from core.domain import topic_services
+from core.platform import models
+from core.tests import test_utils
+import feconf
+import python_utils
+
+(
+    audit_models, exp_models, opportunity_models,
+    user_models
+) = models.Registry.import_models([
+    models.NAMES.audit, models.NAMES.exploration, models.NAMES.opportunity,
+    models.NAMES.user
+])
+
+BOTH_MODERATOR_AND_ADMIN_EMAIL = 'moderator.and.admin@example.com'
+BOTH_MODERATOR_AND_ADMIN_USERNAME = 'moderatorandadm1n'
+
+
+PARAM_NAMES = python_utils.create_enum('test_feature_1')  # pylint: disable=invalid-name
+FEATURE_STAGES = platform_parameter_domain.FEATURE_STAGES
+
+
+class AndroidConfigTest(test_utils.GenericTestBase):
```

Comment:
We should also test that the correct structures are being created & populated. Consider all the things that need to go right in the controller in order for the Android end-to-end tests to pass--all of these things should be validated in the tests below to ensure unexpected failures are caught before they reach the Android test suite.

Reply:
Added most of the validators required and also tested the images and the thumbnails,
Let me know If I've missed anything specific here

Reply:
I think these are much more thorough--thanks!